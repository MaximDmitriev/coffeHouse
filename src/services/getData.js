// export default class getData {
    
//     url = process.env.PUBLIC_URL + "/db.json";


// }