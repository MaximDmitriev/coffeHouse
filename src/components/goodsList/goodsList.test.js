import React from 'react';
import GoodsList from "./goodsList";
import {shallow} from "enzyme";
import getData from "../../services/getData";

describe("Testing <GoodsList />", () => {
  const items = shallow(<GoodsList />);
  const data = new getData();

  describe("Testing Shapshots & state",() => {

    it("GoodsList have rendered", () => {
        expect(items).toMatchSnapshot();
    });

    it("GoodsList state 'items' is empty", () => {
      expect(items.state().items).toBeNil();
    });

    it("GoodsList state 'loading' is true", () => {
      expect(items.state().loading).toBeTrue();
    });

    it("GoodsList state 'error' is empty", () => {
      expect(items.state().error).toBeNil();
    });
  });

  describe("Handlers tests", () => {
    it("Testing componentWillMount", () => {

      items.instance().data.getBest()
      .then(() => expect(items.state().loading).toBeFalse())
      .then(res => expect(items.state().items).toBe(res))
      .catch(res =>expect(items.state().error).toBe(res));
    });
  });
});