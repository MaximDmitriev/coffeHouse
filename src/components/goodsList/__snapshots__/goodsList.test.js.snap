// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GoodsList /> Testing Shapshots & state GoodsList have rendered 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GoogsList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Spinner
        style={
          Object {
            "height": "3rem",
            "margin": "70px auto",
            "width": "3rem",
          }
        }
        tag="div"
        type="border"
      >
        Loading...
      </Spinner>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Loading...",
        "style": Object {
          "height": "3rem",
          "margin": "70px auto",
          "width": "3rem",
        },
        "tag": "div",
        "type": "border",
      },
      "ref": null,
      "rendered": "Loading...",
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  flex-wrap: wrap;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Spinner
          style={
            Object {
              "height": "3rem",
              "margin": "70px auto",
              "width": "3rem",
            }
          }
          tag="div"
          type="border"
        >
          Loading...
        </Spinner>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Loading...",
          "style": Object {
            "height": "3rem",
            "margin": "70px auto",
            "width": "3rem",
          },
          "tag": "div",
          "type": "border",
        },
        "ref": null,
        "rendered": "Loading...",
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  flex-wrap: wrap;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
